openhgnn/models/DHNE.py

```diff
@@ -0,0 +1,93 @@
+
+
+import torch.nn as nn
+from openhgnn.models import BaseModel, register_model
+import torch
+
+
+@register_model("DHNE")
+class DHNE(BaseModel):
+    r"""
+        **Title:** Structural Deep Embedding for Hyper-Networks
+
+        **Authors:** Ke Tu, Peng Cui, Xiao Wang, Fei Wang, Wenwu Zhu
+
+        DHNE was introduced in `[paper] <https://arxiv.org/abs/1711.10146>`_
+        and parameters are defined as follows:
+
+        Parameters
+        ----------
+        dim_features : array
+            the embedding dimension of nodes
+        embedding_sizes : int
+            the embedding dimension size
+        hidden_size : int
+            The hidden full connected layer size
+        nums_type : list
+            the type of nodes
+        device : int
+            the device DHNE working on
+        encodeds :
+            the input of autoencoder (the first layer of DHNE)
+        decodeds :
+            the output of autoencoder (the first layer of DHNE)
+        hidden_layer:
+            the second layer of DHNE
+        ouput_layer :
+            the third layer of DHNE
```

Comment:
The Parameters listed here should correspond to the parameters in the function "__init__". The encodeds, decodeds, hidden_layer, and output_layer should be listed in the descriptions or deleted.